### PR TITLE
Add `sbt-tpolecat` to prevent common bug patterns

### PR DESCRIPTION
**Pull request by froth**
See also: https://github.com/typelevel/cats-effect/pull/4054/

~~also update sbt to run with newer JVMs
(https://github.com/sbt/sbt/issues/7235)~~

### CODE REVIEW

**Comment by TonioGela** [comment]
Hello @froth, thanks for your contribution.
What a timing! I had created [this branch here](https://github.com/typelevel/ce3.g8/compare/main...TonioGela:ce3.g8:fix-build) yesterday evening that fixes many issues in the build, sbt version, GitHub workflow and most lib versions.
There are overlaps between our branches, but I'm more prone to open a PR from mine to fix more things at the same time. 
What I can offer you, if you like the idea is to rework this PR to add this scalacOption but via the addition of [sbt-tpolecat](https://github.com/typelevel/sbt-tpolecat) to the build, WDYT?

**Comment by froth** [comment]
@TonioGela timing indeed :) no worries, I can redo this change onto your branch once merged.

I am not sure about `sbt-tpolecat` for this template however. I am a huge fan of the plugin and use it myself. However in the default configuration it treats warnings as errors. This could lead to frustration for people just wanting to try out cats effect. They would have to remove all unused imports before trying out their code etc. What do you think?

**Comment by froth** [comment]
> * Setup as the default mode [the verbose one](https://github.com/typelevel/sbt-tpolecat?tab=readme-ov-file#verbose-mode)

I think the underlying question is: what is the intended audience of this template. People wanting to try ce out or people bootstrapping production code(with CI, etc). As always the answer is "both" but I personally lean more towards the "trying out" usecase (my personal experience is that in $WORK related projects most people build their own templates anyways). Therefore I  think that verbose mode and comment is a good compromise.

**Comment by froth** [comment]
> Once the PR is ready (this one or a new one that you may want to open), we can share the link in Discord and discuss it a bit before making any decision.

That sounds like a good solution. 
I will reset this one to preserve our discussion. And I will have an eye on discord :)

File: src/main/g8/build.sbt [hunk 1]
```diff
@@ -1,5 +1,8 @@
+import org.typelevel.sbt.tpolecat.*
+
 ThisBuild / organization := "com.example"
 ThisBuild / scalaVersion := $if(scala3.truthy)$"3.4.0"$else$"2.13.13"$endif$
+ThisBuild / tpolecatDefaultOptionsMode := VerboseMode;
```
Comment (TonioGela) [inline src/main/g8/build.sbt:5]: Can we add here the aforementioned comment? Something like `Remember to set "SBT_TPOLECAT_CI" in your CI to enable fatal-warnings and yada yada`?

Comment (froth) [reply src/main/g8/build.sbt:5]: ah sorry, my bad

Comment (TonioGela) [reply src/main/g8/build.sbt:7]:
```suggestion
ThisBuild / tpolecatDefaultOptionsMode := VerboseMode
```

**Comment by BalmungSan** [comment]
My two cents, just add **sbt-tpolecat**, it makes the management of all those flags pretty easy and **Scala** version resistant. Also, in general, most of the flags are useful for writing good **CE** apps.
If people find the warnings too cumbersome then they may tweak them. Adding info about the plugin being used and how to manage it, or a link to the plugin docs would be great.

Note however that, I am biased since I always use the plugin and I have never ever bothered to use the _"development"_ mode.

**Comment by TonioGela** [comment]
So, a lot of opinions have been expressed on Discord, and it seems that the leading opinion is [prompting the user whether to add fatal-warnings or not](https://discord.com/channels/632277896739946517/632277897448652844/1230265895482429522) with sbt-tpolecat always included. 
@froth do you think you can rework the PR to add this flag maybe?

**Comment by froth** [comment]
@TonioGela can do so. I think g8 wants a default for the option. I would use "without fatal" if that is ok

**Comment by TonioGela** [comment]
> @TonioGela can do so. I think g8 wants a default for the option. I would use "without fatal" if that is ok

I think the "vox populi" is in favor of keeping them enabled :)

**Comment by froth** [comment]
@TonioGela can do so. I think g8 wants a default for the option. I would use "without fatal" as default, ok?

**Comment by BalmungSan** [comment]
Yeah it seems folks would prefer without fatal for newcomers.

Also, does the template generate a `README`? If so, it would be great to add a note about the plugin there.

**Comment by TonioGela** [comment]
> Yeah it seems folks would prefer without fatal for newcomers.
> 
> Also, does the template generate a `README`? If so, it would be great to add a note about the plugin there.

Yes, let's add a README and a note, lgtm

**Comment by froth** [comment]
Thanks for the feedback :)

I added a paragraph to the README and made fatal warnings configurabel.

**Review by BalmungSan — APPROVED** [review]

**Pull request merged**